src/server.c

```diff
@@ -2027,126 +2022,164 @@ void createSharedObjectsWithCompat(void) {
+        shared.slowscripterr = shared.slowscripterr_redis;
+        shared.slowmoduleerr = shared.slowmoduleerr_redis;
+        shared.bgsaveerr = shared.bgsaveerr_redis;
+    } else {
+        shared.loadingerr = shared.loadingerr_valkey;
+        shared.slowevalerr = shared.slowevalerr_valkey;
+        shared.slowscripterr = shared.slowscripterr_valkey;
+        shared.slowmoduleerr = shared.slowmoduleerr_valkey;
+        shared.bgsaveerr = shared.bgsaveerr_valkey;
+    }
+}
+
 void createSharedObjects(void) {
     int j;
 
     /* Shared command responses */
-    shared.ok = createObject(OBJ_STRING, sdsnew("+OK\r\n"));
-    shared.emptybulk = createObject(OBJ_STRING, sdsnew("$0\r\n\r\n"));
-    shared.czero = createObject(OBJ_STRING, sdsnew(":0\r\n"));
-    shared.cone = createObject(OBJ_STRING, sdsnew(":1\r\n"));
-    shared.emptyarray = createObject(OBJ_STRING, sdsnew("*0\r\n"));
-    shared.pong = createObject(OBJ_STRING, sdsnew("+PONG\r\n"));
-    shared.queued = createObject(OBJ_STRING, sdsnew("+QUEUED\r\n"));
-    shared.emptyscan = createObject(OBJ_STRING, sdsnew("*2\r\n$1\r\n0\r\n*0\r\n"));
-    shared.space = createObject(OBJ_STRING, sdsnew(" "));
-    shared.plus = createObject(OBJ_STRING, sdsnew("+"));
+    shared.ok = makeObjectShared(createObject(OBJ_STRING, sdsnew("+OK\r\n")));
+    shared.emptybulk = makeObjectShared(createObject(OBJ_STRING, sdsnew("$0\r\n\r\n")));
+    shared.czero = makeObjectShared(createObject(OBJ_STRING, sdsnew(":0\r\n")));
+    shared.cone = makeObjectShared(createObject(OBJ_STRING, sdsnew(":1\r\n")));
+    shared.emptyarray = makeObjectShared(createObject(OBJ_STRING, sdsnew("*0\r\n")));
+    shared.pong = makeObjectShared(createObject(OBJ_STRING, sdsnew("+PONG\r\n")));
+    shared.queued = makeObjectShared(createObject(OBJ_STRING, sdsnew("+QUEUED\r\n")));
+    shared.emptyscan = makeObjectShared(createObject(OBJ_STRING, sdsnew("*2\r\n$1\r\n0\r\n*0\r\n")));
+    shared.space = makeObjectShared(createObject(OBJ_STRING, sdsnew(" ")));
+    shared.plus = makeObjectShared(createObject(OBJ_STRING, sdsnew("+")));
 
     /* Shared command error responses */
-    shared.wrongtypeerr =
-        createObject(OBJ_STRING, sdsnew("-WRONGTYPE Operation against a key holding the wrong kind of value\r\n"));
-    shared.err = createObject(OBJ_STRING, sdsnew("-ERR\r\n"));
-    shared.nokeyerr = createObject(OBJ_STRING, sdsnew("-ERR no such key\r\n"));
-    shared.syntaxerr = createObject(OBJ_STRING, sdsnew("-ERR syntax error\r\n"));
-    shared.sameobjecterr = createObject(OBJ_STRING, sdsnew("-ERR source and destination objects are the same\r\n"));
-    shared.outofrangeerr = createObject(OBJ_STRING, sdsnew("-ERR index out of range\r\n"));
-    shared.noscripterr = createObject(OBJ_STRING, sdsnew("-NOSCRIPT No matching script.\r\n"));
+    shared.wrongtypeerr = makeObjectShared(createObject(
+            OBJ_STRING, sdsnew("-WRONGTYPE Operation against a key holding the wrong kind of value\r\n")));
```

Comment:
If we anyway touch these lines, maybe we can add a small function `createSharedString` to do this. It makes the code smaller:

```suggestion
    shared.wrongtypeerr = createSharedString("-WRONGTYPE Operation against a key holding the wrong kind of value\r\n");
```

The implementation can be static inline, very small:

```c
static inline robj *createSharedString(const char *string) {
    return makeObjectShared(createObject(OBJ_STRING, sdsnew(string));
}

Reply:
it bother me there are some different usage:
```
    shared.wrongtypeerr = makeObjectShared(createObject(
	            OBJ_STRING, sdsnew("-WRONGTYPE Operation against a key holding the wrong kind of value\r\n")));

shared.messagebulk = makeObjectShared(createStringObject("$7\r\nmessage\r\n", 13));

        shared.mbulkhdr[j] = makeObjectShared(createObject(OBJ_STRING, sdscatprintf(sdsempty(), "*%d\r\n", j)));
```

Use createStringObject seem better since it can handle OBJ_ENCODING_EMBSTR. I want to unify them, but i also don't want to touch them on the other side.